### DISCUSSION TITLE
Can't get prerender() documentProps to work (to set page title)

### DISCUSSION THREAD

**Discussion by Niocas** [vikejs/vike#994] [Discussion] [answered]
I have been trying to set the documentProps inside the prerender() function inside my "index.page.server.tsx" - but it seems impossible.
I tried the example from GitHub, it did not work. I looked inside the documentation and nothing worked. Even the example for "npm run dev" with onBeforeRender had problems and I came up with this workaround. I dont know what to do anymore, I tried so many things in prerender, but they always end up being undefined, I am using prerender: true in my vite config and running the server without prerending works fine. Please help me:

- Using ReactJS, Typescript, vite-plugin-ssr, tailwindcss

OnBeforeRender Workaround (works for npm run dev):
![image](https://github.com/brillout/vite-plugin-ssr/assets/75745964/51d36dae-9f98-4c64-a5d6-3c76ac916279)

Prerender part, doesnt work at all, also tried the export variant here:
![image](https://github.com/brillout/vite-plugin-ssr/assets/75745964/d03e7d77-783f-4d7b-a8fb-1c4cb6655513)

**Comment by Niocas** [comment, marked as answer]
![image](https://github.com/brillout/vite-plugin-ssr/assets/75745964/bac7d379-12c8-4db6-a438-8ae4210f4df8)

Never mind getDocumentProps worked after some adjustments:
https://vite-plugin-ssr.com/head

**Reply by brillout** [reply]
(FYI `vike-react` will have built-in support for basic things like this, which will make it easier to get started.)
